core/src/main/scala/org/apache/spark/deploy/history/HistoryAppStatusListener.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.history
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.config.History._
+import org.apache.spark.scheduler.SparkListenerExecutorAdded
+import org.apache.spark.scheduler.cluster.ExecutorInfo
+import org.apache.spark.status.{AppStatusListener, AppStatusSource, ElementTrackingStore}
+
+private[spark] class HistoryAppStatusListener(
+    kvstore: ElementTrackingStore,
+    conf: SparkConf,
+    live: Boolean,
+    appStatusSource: Option[AppStatusSource] = None,
+    lastUpdateTime: Option[Long] = None)
+  extends AppStatusListener(kvstore, conf, live, appStatusSource, lastUpdateTime) {
+
+  override def onExecutorAdded(event: SparkListenerExecutorAdded): Unit = {
+    val execInfo = event.executorInfo
+    val newExecInfo = new ExecutorInfo(execInfo.executorHost, execInfo.totalCores,
+      renewLogUrls(execInfo), execInfo.attributes)
+
+    super.onExecutorAdded(event.copy(executorInfo = newExecInfo))
+  }
+
+  def renewLogUrls(execInfo: ExecutorInfo): Map[String, String] = {
```

Comment:
nit: rename to `replaceLogUrls` (or something like that)

Reply:
`replaceLogUrls` sounds better. Will address.